docs/index.asciidoc

```diff
@@ -0,0 +1,264 @@
+:plugin: google_cloud_storage
+:type: input
+:default_codec: plain
+
+///////////////////////////////////////////
+START - GENERATED VARIABLES, DO NOT EDIT!
+///////////////////////////////////////////
+:version: %VERSION%
+:release_date: %RELEASE_DATE%
+:changelog_url: %CHANGELOG_URL%
+:include_path: ../../../../logstash/docs/include
+///////////////////////////////////////////
+END - GENERATED VARIABLES, DO NOT EDIT!
+///////////////////////////////////////////
+
+[id="plugins-{type}s-{plugin}"]
+
+=== Google Cloud Storage Input Plugin
+
+include::{include_path}/plugin_header.asciidoc[]
+
+
+==== Description
+
+Extracts events from files in a Google Cloud Storage bucket.
+
+Example use-cases:
+
+ * Read https://cloud.google.com/stackdriver/[Stackdriver logs] from a Cloud Storage bucket into Elastic.
+ * Read gzipped logs from cold-storage into Elastic.
+ * Restore data from an Elastic dump.
+ * Extract data from Cloud Storage, transform it with Logstash and load it into BigQuery.
+
+Note: While this project is partially maintained by Google, this is not an official Google product.
+
+
+==== Metadata Attributes
+
+The plugin exposes several metadata attributes about the object being read.
+You can access these later in the pipeline to augment the data or perform conditional logic.
+
+[cols="<,<,<",options="header",]
+|=======================================================================
+| Key | Type | Description
+| `[@metadata][gcs][bucket]` | `string` | The name of the bucket the file was read from.
+| `[@metadata][gcs][name]` | `string` | The name of the object.
+| `[@metadata][gcs][metadata]` | `object` | A map of metadata on the object.
+| `[@metadata][gcs][md5]` | `string` | MD5 hash of the data. Encoded using base64.
+| `[@metadata][gcs][crc32c]` | `string` | CRC32c checksum, as described in RFC 4960. Encoded using base64 in big-endian byte order.
+| `[@metadata][gcs][generation]` | `long` | The content generation of the object. Used for object versioning
+| `[@metadata][gcs][line]` | `long` | The position of the event in the file. 1 indexed.
+| `[@metadata][gcs][line_id]` | `string` | A deterministic, unique ID describing this line. This lets you do idempotent inserts into Elasticsearch.
+|=======================================================================
+
+More information about object metadata can be found in the
+https://cloud.google.com/storage/docs/json_api/v1/objects[official documentation].
+
+
+==== Example Configurations
+
+===== Basic
+
+Basic configuration to read JSON logs every minute from `my-logs-bucket`.
+For example, https://cloud.google.com/stackdriver/[Stackdriver logs].
+
+[source,ruby]
+----------------------------------
+input {
+  google_cloud_storage {
+    interval => 60
+    bucket_id => "my-logs-bucket"
+    json_key_file => "/home/user/key.json"
+    file_matches => ".*json"
+    codec => "json_lines"
```

Comment:
Do not get output with this codec, probably this doesn't work because of line.chomp? https://github.com/logstash-plugins/logstash-input-google_cloud_storage/pull/1/files#diff-a10ed0af350ff953f0040542399af1c9R39 -- easy workarounds but I did bash my head around for a bit before I found the problem (just use `codec => "json"` or `codec => "plain"` with a json filter plugin)